### PR TITLE
Compound primary keys and indicies

### LINKED ISSUE #2: Add support for compound indexes
Here's a link to Dexie.js's compound index documentation: https://dexie.org/docs/Compound-Index

### PR DESCRIPTION
**Pull request by anlumo**
Fixes #2

### CODE REVIEW

**Comment by devashishdxt** [comment]
Thanks for creating this PR @anlumo. Can you add a few tests?

Please refer this to run the existing tests: https://rustwasm.github.io/docs/wasm-bindgen/wasm-bindgen-test/browsers.html

**Comment by anlumo** [comment]
> Can you add a few tests?

Done!

File: src/index.rs [hunk 1]
```diff
@@ -15,7 +17,17 @@ impl Index {
...
             name: name.to_owned(),
-            key_path: key_path.to_owned(),
+            key_path: vec![key_path.to_owned()],
+            unique: None,
+            multi_entry: None,
+        }
+    }
+
+    /// Creates a new index with given name and compound key path
+    pub fn new_compound<S: ToString>(name: &str, key_path: impl IntoIterator<Item = S>) -> Self {
```
Comment (devashishdxt) [inline src/index.rs:27]: What happens when the `key_path` iterator is empty?

File: src/object_store.rs [hunk 2]
```diff
@@ -8,7 +9,7 @@ use crate::{Error, Index, Result};
 /// An object store builder.
 pub struct ObjectStore {
     pub(crate) name: String,
-    pub(crate) key_path: Option<String>,
+    pub(crate) key_path: Option<Vec<String>>,
```
Comment (devashishdxt) [inline src/object_store.rs:12]: I think you can remove `Option` and directly use `Vec` (and check for empty `Vec`).

File: src/index.rs (hunk 1 above)
Comment (anlumo) [reply src/index.rs:27]: Same thing as if you wouldn't have passed any key_path.

> If empty or not specified, the object store is created without a key path and uses [out-of-line keys](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Basic_Terminology#out-of-line_key).

Comment (devashishdxt) [reply src/index.rs:27]: I'm not super familiar with JavaScript. But, does it make any difference if we pass single value in `keyPath` or an array with one item as `keyPath`? (https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/createIndex)

**Pull request merged**